File: A2Z/go/sorting/1/insertion_sort.go
package sortone

func InsertionSort(input []int) []int {
	for i := 0; i < len(input); i++ {
		j := i
		for j > 0 && input[j] < input[j-1] {
			tmp := input[j-1]
			input[j-1] = input[j]
			input[j] = tmp
			j--
		}
	}

	return input
}

File: A2Z/js/sorting/1/insertion-sort.mjs
class Solution {
  // Please change the array in-place
  insertionSort(arr) {
    for (let i = 0; i < arr.length; i++) {
      let j = i;
      while (j > 0 && arr[j] < arr[j - 1]) {
        const tmp = arr[j - 1];
        arr[j - 1] = arr[j];
        arr[j] = tmp;
        j--;
      }
    }

    return arr;
  }
}

export default Solution;
